### DISCUSSION TITLE
Spotify links yield no data on loadtracks

### DISCUSSION THREAD

**Discussion by michihupf** [TopiSenpai/LavaSrc#124] [Q&A] [answered]
Hi 

I am facing an issue where spotify urls aren't giving me any data, see logs below. 
Queueing the song by just providing the song title and artist as a search query (and letting lavalink do its magic) it finds it on YouTube and plays it completely fine. As far as I understand lavasrc does this exact thing but gets the title and artist by way of the spotify API. Somehow this results in no matches thought. Is there anything I am missing?

Logs:
```
2023-10-15T14:49:58.960+02:00  INFO 7166 --- [  XNIO-1 task-2] l.server.player.AudioLoaderRestHandler   : Got request to load for identifier "https://open.spotify.com/track/4zuZkPo9wVNTmuSWj5DaZk?si=e52817fb4a2149cb"
2023-10-15T14:49:58.960+02:00  INFO 7166 --- [  XNIO-1 task-2] lavalink.server.player.AudioLoader       : No matches found
2023-10-15T14:49:58.964+02:00  INFO 7166 --- [  XNIO-1 task-2] l.server.io.RequestLoggingFilter         : GET /v4/loadtracks?identifier=https%3A%2F%2Fopen.spotify.com%2Ftrack%2F4zuZkPo9wVNTmuSWj5DaZk%3Fsi%3De52817fb4a2149cb, client=127.0.0.1
```

Configuration: `application.yml`
```yml
plugins:
  lavasrc:
    providers:
      - "ytsearch:\"%ISRC%\""
      - "ytsearch:%QUERY%"
    sources:
      youtube: true
    spotify:
      clientId: "REDACTED"
      clientSecret: "REDACTED"
      countryCode: "US"
      playlistLoadLimit: 6
      albumLoadLimit: 6
lavalink:
  plugins:
    - dependency: "com.github.topi314.lavasrc:lavasrc-plugin:4.0.0-beta.7"
      repository: "https://maven.topi.wtf/releases"
```

Relevant versions:
`lavalink 4.0.0-beta.4`
`lavasrc 4.0.0-beta.7`

For context. I am using [lavalink-rs](https://github.com/vicky5124/lavalink-rs) to make a private discord music bot. YouTube and SoundCloud already work but Spotify is giving me some issues.

**Comment by topi314** [comment, marked as answer]
well, perhaps you should enable spotify?

**Reply by michihupf** [reply]
Oh did not know it had to be enabled in sources as well. I thought that was just for searching via spotify. But now that I think about it that wouldn't have made sense anyway, as spotify is not used to stream the audio. Anyway that seemed to fix it. Apologies for the inconvenience.
